formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/StmtConversionContext.kt

```diff
@@ -253,7 +254,8 @@ fun StmtConversionContext.convertMethodWithBody(
     // as we may not encounter any `return` statement in the body
     returnTarget.variable.withIsUnitInvariantIfUnit().toViperUnusedResult(linearizer)
 
-    body.checkValidity(declaration.source, errorCollector)
+    if (!body.allNodesSatisfyPurityRequirements(declaration.source, errorCollector)) return null
```

Comment:
This is not a reasonable implementation; if we consider this an error, we should stop conversion, not just generate a dummy.